### PR TITLE
- RouteUrl is removed; - refatored tests and `Router`s.

### LINKED ISSUE #4130: Route url should be removed
[RouteUrl](https://github.com/mapbox/mapbox-navigation-android/blob/0cd17b89896d196c2e65e036645a8d724e81089d/libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/RouteUrl.kt) was created to avoid using `mapbox-directions`(restricting okhttp/retrofit) dependencies into `OnboardRouter`. Since `OffBoard` and `OnBoard` routers were merged into `libnavigation-router` module, this restriction is not relevant and `OnBoard` router can use `mapbox-directions` classes. 

In this way, `RouteUrl` should be removed, and instead of one must be used `mapbox-directions`

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
Remove `RouteUrl`

Closes #4130 

### Changelog

```
<changelog>Fixed bug with missing annotations in reroute/faster route</changelog>
```

### CODE REVIEW

File: libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/offboard/MapboxOffboardRouter.kt [hunk 1]
```diff
@@ -50,7 +50,7 @@ class MapboxOffboardRouter(
         callback: Router.Callback
     ) {
         mapboxDirections = RouteBuilderProvider
-            .getBuilder(accessToken, context, urlSkuTokenProvider)
+            .getBuilder(context, urlSkuTokenProvider)
```
Comment (RingerJK) [inline libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/offboard/MapboxOffboardRouter.kt:53]: the token is removed, it's provided via `routeOptions` next line

File: libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouter.kt [hunk 2]
```diff
@@ -27,10 +30,12 @@ import kotlinx.coroutines.withContext
  * tiles' version, token. Config is provided via [RoutingTilesOptions].
  *
  * @param navigatorNative Native Navigator
+ * @param context application [Context]
  * @param logger interface for logging any events
  */
 class MapboxOnboardRouter(
     private val navigatorNative: MapboxNativeNavigator,
+    private val context: Context,
```
Comment (Guardiola31337) [inline libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouter.kt:38]: This is a property, any chance we could leak the `Context`? I know that should be the `applicationContext` but double-checking here.

File: libnavigation-router/src/main/java/com/mapbox/navigation/route/offboard/RouteBuilderProvider.kt [hunk 3]
```diff
@@ -11,9 +11,8 @@ import com.mapbox.navigation.base.internal.extensions.inferDeviceLocale
 internal object RouteBuilderProvider {
 
     fun getBuilder(
-        accessToken: String,
         context: Context,
-        urlSkuTokenProvider: UrlSkuTokenProvider
+        urlSkuTokenProvider: UrlSkuTokenProvider?
     ): MapboxDirections.Builder =
         MapboxDirections.builder()
             .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
```
Comment (Guardiola31337) [inline libnavigation-router/src/main/java/com/mapbox/navigation/route/offboard/RouteBuilderProvider.kt:18]: `continueStraight(true)` is set to `true` but afterwards is `reset` to `false` if `options.continueStraight()` is not available? Should we remove this or make `MapboxDirections.Builder.routeOptions` `continueStraight` default to `true`?

Comment (Guardiola31337) [reply libnavigation-router/src/main/java/com/mapbox/navigation/route/offboard/RouteBuilderProvider.kt:18]: `roundaboutExits(true)` is set to `true` but afterwards is `reset` to `false` if `options.roundaboutExits()` is not available? Should we remove this or make `MapboxDirections.Builder.routeOptions` `roundaboutExits` default to `true`?

**Review by Guardiola31337 — COMMENTED** [review]
Left some minor comments @RingerJK

File: libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouter.kt (hunk 2 above)
Comment (RingerJK) [reply libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouter.kt:38]: I'd not worry about that, as you mentioned it's appContex, and this approach we use with  `MapboxOffboardRouter` https://github.com/mapbox/mapbox-navigation-android/blob/07fe4ba304cad3438797ea968932a2f9ec1fff21/libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/offboard/MapboxOffboardRouter.kt#L21-L33

File: libnavigation-router/src/main/java/com/mapbox/navigation/route/offboard/RouteBuilderProvider.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigation-router/src/main/java/com/mapbox/navigation/route/offboard/RouteBuilderProvider.kt:18]: totally make sense, but out of current PR. Caught to https://github.com/mapbox/mapbox-navigation-android/issues/4174

**Review by Guardiola31337 — APPROVED** [review]
Thanks for addressing my questions @RingerJK 🚀

**Pull request merged**